src/core/dfn.js

```diff
@@ -65,18 +66,21 @@ function processAsInternalSlot(title, dfn) {
   }
 
   // If it ends with a ), then it's method. Attribute otherwise.
+  const derivedType = title.endsWith(")") ? "method" : "attribute";
   if (!dfn.dataset.dfnType) {
-    dfn.dataset.dfnType = title.endsWith(")") ? "method" : "attribute";
+    dfn.dataset.dfnType = derivedType;
     return;
   }
 
-  // Perform validation on the dfn's type type.
+  // Perform validation on the dfn's type.
   const allowedSlotTypes = ["attribute", "method"];
-  if (!allowedSlotTypes.includes(dfn.dataset.dfnType)) {
-    const msg = docLink`Invalid ${"[data-dfn-type]"} for defining an internal slot.`;
-    const hint = `The only allowed types are: ${codedJoinOr(allowedSlotTypes, {
+  const { dfnType } = dfn.dataset;
+  if (!allowedSlotTypes.includes(dfnType) || derivedType !== dfnType) {
```

Comment:
This check is duplicated from above, given `derivedType` can only be one of `allowedSlotTypes`.
```suggestion
  if (!allowedSlotTypes.includes(dfnType)) {
```

Reply:
the `dfnType` could be "method", but the `derivedType` could be attribute.